### ISSUE TITLE
"Invalid shape for y" when calling fit() on LSTM w/ return_sequences=True (and sklearn API)

### ISSUE THREAD

**Issue by MaxPowerWasTaken** [fchollet/keras#6080]
Full question with reproduceable code example at:
http://stackoverflow.com/questions/43129758/invalid-shape-for-y-for-keras-lstm-w-return-sequences-true-and-sklearn-api

In summary, I have a sequence-classification task where my X and target Y are the same shape (2D rows=locations columns=time intervals). If I reshape my X & Y to be 3D tensors I get an error 

> "ValueError('Invalid shape for y')"

 thrown from the sklearn wrapper that y must be 2-D.

If I reshape X to be a 3D tensor but leave Y as 2-D, I get another error that 

> "model target: expected lstm_17 to have 3 dimensions."

Any help on how to do this sequence-to-sequence classification with the last LSTM layer having return_sequences=True would be much appreciated. 

- [X] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [X] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).
- [X] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by MaxPowerWasTaken** [comment]
A commenter at the Stack Overflow link above says my code example works for him if he switches `loss` to `binary_crossentropy`, though I continue to have the same issue. I updated to latest Keras (2.0.2) to post this issue, it's working for him on Keras 1.2.2

**Comment by pedrobisp** [comment]
Issue still happening. Any thoughts on this?

In my use case, I could flatten the output of my last layer to tackle the issue, but I cannot do it because I also use a Masking layer. My current solution is not using the scikit-learn wrapper since it only accepts 2D tensors for y.

**Comment by afern100** [comment]
I am still having the same problem

**Comment by dmitriyboyuk** [comment]
Same problem, confused here
